pandas/tests/io/formats/test_printing.py

```diff
@@ -3,11 +3,33 @@
 from collections.abc import Mapping
 import string
 
+import pytest
+
 import pandas._config.config as cf
 
+import pandas as pd
+
 from pandas.io.formats import printing
 
 
+@pytest.mark.parametrize(
+    "input_names, expected_names",
+    [
+        (["'a", "b"], ["'a", "b"]),  # Escape leading quote
+        (["test's", "b"], ["test's", "b"]),  # Escape apostrophe
+        (["'test'", "b"], ["'test'", "b"]),  # Escape surrounding quotes
+        (["test", "b'"], ["test", "b'"]),  # Escape single quote
+        (["'test\n'", "b"], ["'test\n'", "b"]),  # Escape and preserve newline
+    ],
+)
+def test_formatted_index_names(input_names, expected_names):
+    # GH#60190
+    df = pd.DataFrame({name: [1, 2, 3] for name in input_names}).set_index(input_names)
+    formatted_names = df.index.names
+
+    assert formatted_names == expected_names
```

Comment:
This test passes on the main branch, so is not testing your changes. You can do `str(df.index.names)` and compare this to the expected value, e.g. `['\'a', 'b']`

Reply:
There is an issue that I didn't realize before, that's why the tests pass. When the tests look like this
![image](https://github.com/user-attachments/assets/1c5e4fc0-b9ef-45ae-9519-5c9ecd816a66)
running the pre-commit checks:  `pre-commit run --from-ref=upstream/main --to-ref=HEAD --all-files` fail, and convert it to the format above (i.e. not escaping the single quotes). Then I didn't realize and made a PR, that's why this happens. Is there a solution to this?

Reply:
When comparing to `str(df.index.names)`, you want the expected to be a single string, not a list of strings. But indeed, you need to escape the backslash as well: `['\\'a', 'b']`.

Reply:
Just to clarify my previous comment, you cannot compare single entries because with a backslash (this PR) and without a backslash (main) evaluate as equal. However if you convert the entire `index.names` to a string, they do not.

```python
input_names = ["'a", "b"]
df = pd.DataFrame({name: [1, 2, 3] for name in input_names}).set_index(input_names)

print(str(df.index.names) == "['\\'a', 'b']")  # main
# False

print(str(df.index.names) == "['\\'a', 'b']")  # PR
# True
```

Reply:
Okay I think I'm with you :)